src/sentry/utils/outcomes.py

```diff
@@ -71,11 +73,40 @@ def track_outcome(
     assert isinstance(category, (type(None), DataCategory))
     assert isinstance(quantity, int)
 
+    use_billing = outcome.is_billing() and outcomes_billing is not None
+
+    # Send billing outcomes to a dedicated topic or even cluster if there is a
+    # separate configuration for it. Otherwise, fall back to the regular
+    # outcomes topic.
+    #
+    # In Sentry, there is no significant difference between the classes of
+    # outcome. In Sentry SaaS, they have elevated stability requirements as they
+    # are used for spike protection and quota enforcement.
+    topic_name = outcomes_billing["topic"] if use_billing else outcomes["topic"]
+
+    # Create a second producer instance only if the cluster differs. Otherwise,
+    # reuse the same producer and just send to the other topic.
+    if use_billing and outcomes_billing["cluster"] != outcomes["cluster"]:
```

Comment:
i forgot to tell you, you can do one thing to deduplicate this, and that is to avoid the use of `global ` entirely

i.e. you can have a global dictionary of outcomes producers keyeed by some string, then you can pull out the common code from both branches intoo a helper fn

Reply:
Thanks, as long as it's just the two I'll stick to separate globals. Avoids typos.